java/wavefrontsender.md

```diff
@@ -5,103 +5,158 @@
+### Prerequisites
+
+Before your application can use a `WavefrontProxyClient`, you must [set up and start a Wavefront proxy](https://docs.wavefront.com/proxies_installing.html).
+
+### Initialize the WavefrontClient
 
-When sending data via the Wavefront proxy, you need to create a `WavefrontProxyClient`. Include the following information.
+You initialize a `WavefrontClient` as shown in the examples below.
 
-* The name of the host that will run the Wavefront proxy.
-* One or more proxy listening ports to send data to. The ports you specify depend on the kinds of data you want to send (metrics, histograms, and/or trace data). You must specify at least one listener port. 
-* Optional settings for tuning communication with the proxy.
+Optionally, you can call factory methods to tune the following ingestion properties:
+
+* Max queue size - Internal buffer capacity of the `WavefrontSender`. Data that exceeds this size is dropped.
+* Flush interval - Interval for flushing data from the `WavefrontSender` directly to Wavefront.
+* Batch size - Amount of data to send to Wavefront in each flush interval.
 
-> **Note**: See [Advanced Proxy Configuration and Installation](https://docs.wavefront.com/proxies_configuring.html) for details.
+Together, the batch size and flush interval control the maximum theoretical throughput of the `WavefrontSender`. 
+> **Note**: Override the defaults values only when you want to set higher values.
+
+**Example**: Use a factory class to create a WavefrontClient and send data to Wavefront via Wavefront Proxy. 
 
 ```java
-// Create the builder with the proxy hostname or address
-WavefrontProxyClient.Builder wfProxyClientBuilder = new WavefrontProxyClient.Builder(proxyHostName);
+// Add a client with the following URL format: "proxy://<your.proxy.load.balancer.com>:<port>" or "http://<host>:<port>/"
+// to send data to proxies
+WavefrontClientFactory wavefrontClientFactory = new WavefrontClientFactory();
+wavefrontClientFactory.addClient(wavefrontURL);
 
-// Set the proxy port to send metrics to. Default: 2878
-wfProxyClientBuilder.metricsPort(2878);
+WavefrontSender wavefrontSender = wavefrontClientFactory.getClient();
+```
+**Example**: Use a builder to create a WavefrontClient and send data to Wavefront via Wavefront Proxy. 
 
-// Set a proxy port to send histograms to.  Recommended: 2878
-wfProxyClientBuilder.distributionPort(2878);
+```java
+// Using the WavefrontClient.Builder directly with a url in the form of "http://<your.proxy.load.blanacer>:<port>"
+// to send data to proxies.
+WavefrontClient.Builder wfClientBuilder = new WavefrontClient.Builder(proxyURL);
 
-// Set a proxy port to send trace data to. Recommended: 30000
-wfProxyClientBuilder.tracingPort(30_000);
+//The maximum message size in bytes that is pushed with on each flush interval 
+wfClientBuilder.messageSizeBytes(120);
```

Comment:
I don't think we need to list this builder method because it's more of an edge use case (when size of payload should not exceed a certain threshold).  It's also a bit confusing in general what this is in relation to batchSize.  

But if we do include it, it should be listed last, and with a larger size as the example (like 1MB = 1_048_576)